lib/core/src/Cardano/Wallet/Primitive/AddressDerivation.hs

```diff
@@ -707,3 +711,20 @@ hex = convertToBase Base16
 -- | Decode a 'ByteString' from base16
 fromHex :: ByteArray bout => ByteString -> Either String bout
 fromHex = convertFromBase Base16
+
+-- | Convert a key as hex-encoded text
+keyToHexText :: XPrv -> Text
+keyToHexText =
```

Comment:
This is a very misleading name for a function that actually modifies the internals of the `XPrv`! 
Plus, this function is trying to do 2 things at once:

- Stripping the public part
- Converting to hex

We already have a function that convert ByteString to hexadecimal, make use of it! It's declared 10 lines above!
Now, what about writing a function `stripPubKey :: ByteString -> ByteString` and at call-site, compose them all!

```hs
let bytes = hex . stripPubKey . unXPrv $ xprv 
```

Much clearer and honest about what's being done!

Reply:
👍 Went with `unXPrvStripPub` which you mentioned as an alternative in DM

>  compose them all!

`stripPubKey` is a bit too composable for my liking in this case. `stripPubKey . stripPubKey` makes no sense.

Here I don't see any benefit of composability. In fact, it would also be nice to be keep the jcli-compatibility test as similar to the actual usage in the CLI as possible.